File: artwalk_backend/src/main/java/com/ssafy/a401/artwalk_backend/domain/user/repository/UserRepository.java
package com.ssafy.a401.artwalk_backend.domain.user.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import com.ssafy.a401.artwalk_backend.domain.user.model.User;

@Repository
public interface UserRepository extends JpaRepository<User, String> {
	int countByRefreshToken(String refreshToken);
}

File: Artwalk_Android/domain/src/main/java/com/a401/domain/repository/UserRepository.kt
package com.a401.domain.repository

import kotlinx.coroutines.flow.Flow
import com.a401.domain.model.User

interface UserRepository {

    suspend fun postIdToken(idToken: String, serviceType: String): Flow<String>

    suspend fun postLoginInfo(userId: String, password: String): Flow<String>

    suspend fun postLogin(accessToken: String, refreshToken: String)

    suspend fun postRegist(user: User, password: String): Flow<String>

    suspend fun getUserInfo(): Flow<User>
}